lite-rpc/src/main.rs

```diff
@@ -79,13 +75,53 @@ pub async fn main() -> anyhow::Result<()> {
         clean_interval_ms,
         enable_postgres,
         prometheus_addr,
-    );
+    )
+    .await
+}
+
+fn get_args() -> Args {
+    let mut args = Args::parse();
+
+    dotenv().ok();
+
+    args.enable_postgres = args.enable_postgres
+        || if let Ok(enable_postgres_env_var) = env::var("PG_ENABLED") {
+            enable_postgres_env_var != "false"
+        } else {
+            false
+        };
+
+    args
+}
+
+#[tokio::main(flavor = "multi_thread", worker_threads = 16)]
+pub async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::init();
+
+    let args = get_args();
+
+    let main: AnyhowJoinHandle = tokio::spawn(async move {
+        loop {
+            let Err(err) = start_lite_rpc(args.clone()).await else{
+                bail!("LiteBridge services returned without error");
+            };
+
+            // log and restart
+            log::error!("Services quit unexpectedly {err:?} restarting in {RESTART_DURATION:?}");
```

Comment:
It will log error here

Reply:
Have you tested if it is working?

Reply:
Yes. Earlier it didn't log cause it was bailing out from the main. Which for some reason is not captured by Prometheus.

Reply:
Ok, lets see then on fly if it works.